phoenix-core/src/main/java/org/apache/phoenix/execute/ClientAggregatePlan.java

```diff
@@ -183,13 +198,15 @@ public ExplainPlan getExplainPlan() throws SQLException {
         if (where != null) {
             planSteps.add("CLIENT FILTER BY " + where.toString());
         }
-        if (!groupBy.isEmpty()) {
-            if (!groupBy.isOrderPreserving()) {
-                planSteps.add("CLIENT SORTED BY " + groupBy.getKeyExpressions().toString());
-            }
+        if (groupBy.isEmpty()) {
+            planSteps.add("CLIENT AGGREGATE INTO SINGLE ROW");
+        } else if (groupBy.isOrderPreserving()) {
             planSteps.add("CLIENT AGGREGATE INTO DISTINCT ROWS BY " + groupBy.getExpressions().toString());
+        } else if (useHashAgg) {
+            planSteps.add("CLIENT HASH AGGREGATE INTO DISTINCT ROWS BY " + groupBy.getExpressions().toString());
```

Comment:
Add  CLIENT SORTED BY line here if sorting required for hash aggregate.

Reply:
Done.